packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/dataservice/ModelController.java

```diff
@@ -80,6 +97,28 @@ ResponseEntity<Model> getModel(
 			return ResponseEntity.noContent().build();
 		}
 
+		// Find the Document Assets linked via provenance to the model
+		ProvenanceQueryParam body = new ProvenanceQueryParam();
+		body.setRootId(id);
+		body.setRootType(ProvenanceType.MODEL);
+		body.setTypes(List.of(ProvenanceType.DOCUMENT));
+		final JsonNode provenanceResults = provenanceProxy.search(body, "models_from_document").getBody();
+		final JsonNode resultsNode = provenanceResults.get("result");
+
+		// If there are results, fetch the Document Assets, gather their extractions
+		if (resultsNode != null && resultsNode.isArray() && !resultsNode.isEmpty()) {
+			final List<String> documentIds = mapper.convertValue(resultsNode, new TypeReference<List<String>>() {});
+			final List<JsonNode> extractions = new ArrayList<>();
+			documentIds.forEach(documentId -> {
+				final DocumentAsset document = documentProxy.getAsset(documentId).getBody();
```

Comment:
You may need to put this in a try/catch, because if this call throws an exception (or if any of these calls do) then i believe you'll be kicked out of the `forEach` loop. If you catch the exception (and just log it?) then you can try again for the next documentid